src/gradualizer_int.erl

```diff
@@ -103,6 +103,9 @@ negate_int_type(RangeTy) ->
 
 %% Integer type to range.
 -spec int_type_to_range(type()) -> int_range().
+%% TODO: `{var, _, _}' should probably generate a constraint on the type var, too,
+%% but fixing a crash is more important at this point.
+int_type_to_range({var,  _, _})                        -> {neg_inf, pos_inf};
```

Comment:
I don't think we should ever get here for `var`, just like we never get here for `any()`. We should make sure we only have an integer type when we get to this module.

Instead, I think the problem should be fixed in `typechecker:negate_num_type/2`, by adding var to the guards in the first clause:

```Erlang
negate_num_type({type, _, TyName, []} = Ty, _Env) when
      TyName =:= any;
      TyName =:= integer;
      TyName =:= float ->
    Ty;
```

Reply:
This function is absent in the stacktrace due to the stack frame being overwritten for tail calls, but it seems to be the only path where this is called and it makes sense for the stack trace posted in the issue IMO.

Reply:
Indeed, adding a clause here works :+1: